hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/OnDemandContainerReplicationSource.java

```diff
@@ -61,7 +69,8 @@ public void copyData(long containerId, OutputStream destination)
     }
 
     controller.exportContainer(
-        container.getContainerType(), containerId, destination, packer);
+        container.getContainerType(), containerId, destination,
+        packer.get(compression));
```

Comment:
Ok. Here we need to take care of exceptions: Can you add a check to ensure compression string is expected and throw exception when it's not?

+1 after this final change.

Reply:
finished